src/seasidefilteredmodel.cpp

```diff
@@ -703,7 +710,10 @@ SeasidePerson *SeasideFilteredModel::personByRow(int row) const
     if(row < 0 || row >= m_contactIds->size()) {
         return NULL;
     }
-    return personFromItem(SeasideCache::itemById(m_contactIds->at(row)));
+    if(m_allContactIds->contains(row)) {
```

Comment:
This is a test for presence of the implicitly-converted value, not index validity.  Also, should be m_contactIds, not m_allContactds (and index validity has already been tested...)